### PR TITLE
ci(mergify): don't add co-author if there's only one author

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

To mitigate OCD.

Originally I wanted to remove the "first-author" (i.e., the author of the squashed commit) from the co-authors, but it seems not possible -- They will be the GitHub PR's author instead of information in git (I guess)!

So I made just a little effort to remove co-author when they are the only author, which is the weirdest situation. 😇

## Additional information

Can play here: http://jinja.quantprogramming.com/

with
```jinja
{{body}}
{% for user in approved_reviews_by %}
Approved-By: {{user}}
{%- endfor %}
{%- set commit_authors = commits | rejectattr("author", "eq", "mergify[bot]") | unique(false, "email_author")| list -%}
{% if commit_authors | length > 1 %}
{% for commit in commit_authors %}
Co-Authored-By: {{ commit.author }} <{{ commit.email_author }}>
{%- endfor %}
{%- endif %}
```
and
```yaml
commits:
  - author: a
    email_author: b
  - author: a1
    email_author: b1
  - author: mergify[bot]
    email_author: 0
approved_reviews_by:
  - 1
  - 2
body: 1
```

### CODE REVIEW

**Review by BugenZhao — APPROVED** [review]
rubber stamp

**Review by TennyZhuang — APPROVED** [review]

File: .mergify.yml [hunk 1]
```diff
@@ -3,13 +3,15 @@ shared:
       {{title}} (#{{number}})
       
       {{body | get_section("## What's changed and what's your intention?")}}
-      
       {% for user in approved_reviews_by %}
       Approved-By: {{user}}
```
Comment (BugenZhao) [inline .mergify.yml:7]: Can we add an `@` before the username?

Comment (xxchan) [reply .mergify.yml:7]: Will the `@` lead to noisy notification? 🥵

Just tested, did you receive any notification?

image

Comment (BugenZhao) [reply .mergify.yml:7]: Good catch. 🤣

|
image

**Comment by xxchan** [comment]
who can hit  the buttion 🤪

**Comment by BugenZhao** [comment]
image
Mergify is back! 😄

**Comment by xxchan** [comment]
> Mergify is back! 😄

These PR are merged by @lmatz , not mergify 😄

**Pull request merged**